presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -132,13 +133,15 @@ public Visitor(Metadata metadata, Session session, List<Expression> parameters,
         @Override
         protected Node visitShowStats(ShowStats node, Void context)
         {
-            validateShowStats(node);
             checkState(queryExplainer.isPresent(), "Query explainer must be provided for SHOW STATS SELECT");
 
             if (node.getRelation() instanceof TableSubquery) {
-                QuerySpecification specification = (QuerySpecification) ((TableSubquery) node.getRelation()).getQuery().getQueryBody();
+                Query query = ((TableSubquery) node.getRelation()).getQuery();
+                QuerySpecification specification = (QuerySpecification) query.getQueryBody();
+                Plan plan = queryExplainer.get().getLogicalPlan(session, new Query(Optional.empty(), specification, Optional.empty(), Optional.empty()), parameters);
+                validateShowStats(node, query, specification, plan);
```

Comment:
Now that this method is used only for a subquery, perhaps, it makes sense to rename it to `validateShowStatsSubquery`